### PR TITLE
Chore/improved build screen

### PR DESCRIPTION
**Pull request by DominikGuzei**
Ups we did it again, we did way too much - this branch should have been small, but it got way to big!

Sets up the new architecture and basic improved build screen.

Have fun, reviewing :-D

### CODE REVIEW

File: app/assets/javascripts/slotcars/build/build_screen_state_manager.js.coffee [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (stravid) [inline app/assets/javascripts/slotcars/build/build_screen_state_manager.js.coffee]: Delete.

File: app/assets/javascripts/slotcars/build/controllers/builder_controller.js.coffee [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  drawController: null
+  buildScreenView: null
+
+  init: ->
+    @track = slotcars.shared.models.TrackModel.createRecord()
+
+    @drawController = slotcars.build.controllers.DrawController.create
+      track: @track
+
+    @drawView = slotcars.build.views.DrawView.create
```
Comment (DominikGuzei) [inline app/assets/javascripts/slotcars/build/controllers/builder_controller.js.coffee]: make drawView private _drawView

File: spec/javascripts/unit/slotcars/build/build_screen_state_manager_spec.js.coffee [hunk 3]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (stravid) [inline spec/javascripts/unit/slotcars/build/build_screen_state_manager_spec.js.coffee]: Delete.

File: app/assets/javascripts/slotcars/shared/views/track_view.js.coffee [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  ROAD_WIDTH: 70
+  SIDE_WIDTH: 10
+  DASH_WIDTH: 2
+  SIDE_DASH_WIDTH: 3
+  SIDE_WIDTH: 5
+
+  didInsertElement: ->
+    @_paper = Raphael @$()[0], 1024, 768
+    
+  redrawTrack: ->
```
Comment (DominikGuzei) [inline app/assets/javascripts/slotcars/shared/views/track_view.js.coffee]: rename to drawTrack

File: app/assets/javascripts/slotcars/slotcars_application.js.coffee [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#= require helpers/namespace
+#= require slotcars/route_manager
+
+namespace 'slotcars'
+
+slotcars.SlotcarsApplication = Ember.Application.extend
+  screenFactory: null
+  currentScreen: null
```
Comment (DominikGuzei) [inline app/assets/javascripts/slotcars/slotcars_application.js.coffee]: make currentScreen private

File: spec/javascripts/unit/slotcars/slotcars_application_spec.js.coffee [hunk 6]
```diff
@@ -0,0 +1,80 @@
+#= require slotcars/slotcars_application
+#= require slotcars/route_manager
+
+describe 'screen managment', ->
+
+  SlotcarsApplication = slotcars.SlotcarsApplication
+
+  describe 'interaction with screens', ->
```
Comment (DominikGuzei) [inline spec/javascripts/unit/slotcars/slotcars_application_spec.js.coffee:8]: could be prettified with new mocking helpers

Comment (DominikGuzei) [inline spec/javascripts/unit/slotcars/slotcars_application_spec.js.coffee:8]: nope :-D

**Pull request merged**